### DISCUSSION TITLE
[Task]: How to inference fastflow model by using pt file? (anomalib 0.3.3)

### DISCUSSION THREAD

**Discussion by jeaho322** [open-edge-platform/anomalib#1894] [Q&A]
### What is the motivation for this task?

Hello, I use fastflow model (anomalib 0.3.3) for work. (can't update to lastest version)
Until now, For fast inference speed, 

1. After traning, I made lightning model ('.ckpt')  to torch model ( '.pt'  ) like below

    logger.info("Training the model.")
    trainer.fit(model=model, datamodule=datamodule)

    logger.info("Loading the best model weights.")
    load_model_callback = LoadModelCallback(weights_path=trainer.checkpoint_callback.best_model_path)
    trainer.callbacks.insert(0, load_model_callback)

    best_model =model.load_from_checkpoint(trainer.checkpoint_callback.best_model_path)
    torch.save(best_model.state_dict(), 'best_model.pt')

2. And I tried to inference, an error occured (in anomalib/tools/inference/torch_inference.py)
    weights ='best_model.pt'
    inferencer = TorchInferencer(path=weights, device= 'cuda')

   KeyError("``model`` is not found in the checkpoint. Please check the checkpoint file.")

3. I saw a solution about config file with export mode but there is no paramter in my config file(anomalib 0.3.3)

### Describe the solution you'd like

1. How I can convert lightning '.ckpt' to torch '.pt' file and infence by torch '.pt' file?

2. If there is better way to improve inference speed, please tell me

In Advance, thank you for your help!!

**Comment by Sj-Yuan** [comment]
you just need to change the "export_mode" in config.yaml about fastflow model.

**Comment by Sj-Yuan** [comment]
And I also want to know how to improve the inference speed.

**Comment by shiyongde** [comment]
how to speedup？？

**Comment by samet-akcay** [comment]
You could use OpenVINO inference to speed it up on CPU. As @Sj-Yuan mentioned, you need to change the `export_mode` to openvino
